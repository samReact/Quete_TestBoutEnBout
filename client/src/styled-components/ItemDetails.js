import styled from 'styled-components';
import { Name as ItemName, Price as ItemPrice } from './Item';

export const Grid = styled.div`
  margin: 128px auto;
  display: grid;
  width: 1024px;
  grid-template-columns: 464px 464px;
  column-gap: 96px;
  padding-inline-start: 0;
`;

export const Name = styled(ItemName)`
  margin-top: 48px;
  text-align: initial;
  font-size: 36px;
`;

export const Price = styled(ItemPrice)`
  margin-top: 16px;
  text-align: initial;
  font-size: 24px;
`;
