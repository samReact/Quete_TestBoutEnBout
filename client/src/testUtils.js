// eslint-disable-next-line import/prefer-default-export
export const items = [
  {
    id: '1234',
    shortName: 'Glouglou',
    inflatedPrice: 100,
    largestImageUrl: 'http://no.big/deal',
  },
  {
    id: '5678',
    shortName: 'mdr',
    largestImageUrl: 'http://okfine.com/lol',
  },
];
