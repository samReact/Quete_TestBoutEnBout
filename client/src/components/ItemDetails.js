import React from 'react';
import PropTypes from 'prop-types';
import ItemImage from './ItemImage';
import ItemPrice from './ItemPrice';
import { Grid, Name, Price } from '../styled-components/ItemDetails';

const ItemDetails = ({ shortName, inflatedPrice, imageUrl }) => (
  <Grid>
    <ItemImage imageUrl={imageUrl} />
    <div>
      <Name>{shortName.toUpperCase()}</Name>
      <ItemPrice
        value={inflatedPrice}
        render={value => <Price>{value}</Price>}
      />
    </div>
  </Grid>
);

ItemDetails.propTypes = {
  shortName: PropTypes.string,
  inflatedPrice: PropTypes.number,
  imageUrl: PropTypes.string,
};

ItemDetails.defaultProps = {
  shortName: '',
  inflatedPrice: null,
  imageUrl: '',
};

export default ItemDetails;
