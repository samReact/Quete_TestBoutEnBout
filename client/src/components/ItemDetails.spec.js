import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../testUtils';
import ItemDetails from './ItemDetails';

describe('ItemDetails', () => {
  describe('when passed no props', () => {
    it('renders without crashing', () => {
      shallow(<ItemDetails />);
      // TODO: cleanup rendering
    });
  });

  describe('when passed item props', () => {
    const props = { ...items[0] };

    it('renders correctly', () => {
      const wrapper = shallow(<ItemDetails {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
