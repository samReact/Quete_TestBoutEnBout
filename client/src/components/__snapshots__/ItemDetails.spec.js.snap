// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails when passed item props renders correctly 1`] = `
<styled.div>
  <ItemImage
    imageUrl=""
  />
  <div>
    <Styled(styled.div)>
      GLOUGLOU
    </Styled(styled.div)>
    <ItemPrice
      render={[Function]}
      value={100}
    />
  </div>
</styled.div>
`;
